src/storage/LockingResourceStore.ts

```diff
@@ -121,11 +124,7 @@ export class LockingResourceStore implements AtomicResourceStore {
    */
   protected async waitForStreamToEnd(source: Readable): Promise<void> {
     try {
-      await new Promise((resolve, reject): void => {
-        source.on('error', reject);
-        source.on('end', resolve);
-        source.on('close', resolve);
-      });
+      await eosPromise(source);
```

Comment:
Maybe just call it something else? Understanding it requires a lookup now. `endOfStream`?